airbyte-webapp/src/components/base/Text/Text.tsx

```diff
@@ -0,0 +1,52 @@
+import classNames from "classnames";
+import React, { useMemo } from "react";
+
+import styles from "./Text.module.scss";
+
+type TextSize = "sm" | "md" | "lg" | "xl";
+type TextElementType = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "span" | "div";
+
+interface TextProps {
+  as?: TextElementType;
```

Comment:
I was inspired by styled components to call this `as` but is the name clear? Alternatively, I thought of `element` or `type` as well. Thoughts?

Reply:
`as` feels fairly standard, though some other opinions would definitely be useful :)

Reply:
I like `as` here

Reply:
++ for `as` as well, as it feels pretty standard as of today.